tools/generator/SourceWriters/MethodCallback.cs

```diff
@@ -79,6 +86,25 @@ protected override void WriteBody (CodeWriter writer)
 
 			if (!method.IsVoid && method.Parameters.HasCleanup)
 				writer.WriteLine ("return __ret;");
+
+			writer.Unindent ();
+
+			writer.WriteLine ("} catch (global::System.Exception __e) {");
+			writer.Indent ();
+			writer.WriteLine ("__envp.SetPendingException (__e);");
```

Comment:
@jpobst: as per [this comment](https://github.com/dotnet/java-interop/issues/1258#issuecomment-2380108625), the generated body should instead be:

```csharp
JniEnvironment.Runtime.OnUnhandledException (ref __envp, __e);
```

The name `OnUnhandledException()` can (should) be debated, in part because the exception is not, in fact, "unhandled".  Perhaps `MarshalExceptionToJni()`?  (Suggestions welcome.)